File: forgerock-auth/src/main/java/org/forgerock/android/auth/callback/BooleanAttributeInputCallback.java
/*
 * Copyright (c) 2019 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.android.auth.callback;

import androidx.annotation.Keep;

import org.json.JSONException;
import org.json.JSONObject;

import lombok.Getter;

/**
 * Implements a Callback for collection of a single identity object attribute from a user.
 */
@Getter
public class BooleanAttributeInputCallback extends AttributeInputCallback {

    /**
     * The attribute Value
     */
    private Boolean value;

    @Keep
    public BooleanAttributeInputCallback() {
    }

    /**
     * Constructor for this Callback.
     */
    @Keep
    public BooleanAttributeInputCallback(JSONObject jsonObject, int index) throws JSONException {
        super(jsonObject, index);
    }


    @Override
    protected void setAttribute(String name, Object value) {
        super.setAttribute(name, value);
        if (VALUE.equals(name)) {
            this.value = (Boolean) value;
        }
    }

    public void setValue(Boolean value) {
        super.setValue(value);
    }

    @Override
    public String getType() {
        return "BooleanAttributeInputCallback";
    }
}

File: e2e/app/src/main/java/com/example/app/callback/BooleanAttributeInputCallback.kt
/*
 * Copyright (c) 2024 - 2025 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.example.app.callback

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import org.forgerock.android.auth.callback.BooleanAttributeInputCallback

@Composable
fun BooleanAttributeInputCallback(callback: BooleanAttributeInputCallback) {

    var input by remember {
        mutableStateOf(callback.value)
    }

    Row(modifier = Modifier
        .padding(4.dp)
        .fillMaxWidth()) {
        Text(text = callback.prompt)
        Spacer(modifier = Modifier.weight(1f, true))
        Switch(
            checked = input,
            onCheckedChange = {
                input = it
                callback.value = it
            }
        )
    }

}